packages/docker-cli/0001-non-tcp-host-header.patch

```diff
@@ -0,0 +1,98 @@
+From c85efebc0c9321fecc536485b47857dbcbe34ddf Mon Sep 17 00:00:00 2001
+From: Sean McGinnis <stmcg@amazon.com>
+Date: Thu, 3 Aug 2023 13:32:10 +0000
+Subject: [PATCH] Backport https://github.com/moby/moby/pull/45942 to vendored
+ docker client
+
+---
```

Comment:
nit: It would be good to include a bit more info in the commit messages about these changes. Just as much to just re-state the problem and solution so one doesn't need to read another GitHub url to understand our git message. 

```
Go 1.20.6 fixed an issue with headers which broke the client code since it didn't set a properly formatted
hostname. This adds a DummyHost for the headers on local communication which fixes local access on builds 
later than Go 1.20.6.
```
This way one knows immediately that we are unlikely to drop this patch moving forward just by reading the patch itself.

Reply:
I copied over the upstream commit's description so we have the full context captured in the patch.